packages/api-client-next/src/index.ts

```diff
@@ -123,7 +136,9 @@ export function createAPIClient<
   ): Promise<RequestReturnType<OPERATION_NAME, OPERATIONS>> {
     const [requestPath, options] = transformPathToQuery(
       pathParam,
-      params?.[0] as Record<string, string>,
+      Array.isArray(params)
+        ? (params?.[0] as Record<string, string>)
+        : (params as Record<string, string>),
```

Comment:
in which case do we have it? Tests needed

Reply:
you know what question to ask. I removed it, and probably it added it due some not related problems I had. cleaned up!